index.html

```diff
@@ -0,0 +1,331 @@
+
+## `@using` within a document
+
+The `@using` directive is applied to `schema` declarations to declare that the schema uses a particular spec. For example,
+
+```graphql
+#::[Using the `@using` directive]
+schema @using(spec: "https://spec.example.com/example/v1.0") {
+  query: Query
+}
+```
+
+## The `@using` directive
+
+```graphql
+#::[Declaration of the `@using` directive]
+directive @using(spec: String!, prefix: String)
+  repeatable on SCHEMA
+```
+
+Documents **must** include a definition for the `@using` directive. The provided definition does not have to match the above definition exactly. Instead, it must be *compatible*. A conforming definition may:
+- **Omit optional arguments** if they are never used in the document.
+- **Introduce new directive locations**, the behavior of which is unspecified here.
+- **Introduce new arguments.** New arguments **should** be prefixed, e.g. `example_feature: Bool`. The prefix **should** be the prefix of a spec referenced with this or another `@using` directive within the document.
+
+### `spec`
+
+A URL identifying the spec. When viewed, the URL **should** provide the content of the appropriate version of the specification in some human-readable form. In short, a human reader should be able to click the link and go to the docs for the version in use. It is not required that the *content* be machine-readable in any particular way.
+
+<figure>
+  <figcaption>
+    <a name=anatomy-of-a-spec-url href=#anatomy-of-a-spec-url class=a-header>Basic anatomy of a spec URL</a>
+  </figcaption>  
+  <code class=anatomy>
+    <span class=pink style='--depth: 2'>http://spec.example.com/a/b/c/<span>exampleSpec<aside>prefix</aside></span><aside>
+      identity
+    </aside></span>/<span style='--depth: 2' class=green>v1<aside>version</aside></span>
+  </code>
+</figure>
+
+Spec URLs contain information about the spec's [prefix](#prefixing) and [version](#versioning), and should be generated and processed in accordance with the [requirements on the structure of spec URLs](#spec-urls).
+
+### `prefix`
+
+Change the [prefix](#prefixing) for schema elements from this specification.
+
+When a `prefix` is provided, processors **must** configure the resulting implementation to replace the default prefix on the names of all [prefixed schema elements](#elements-which-should-be-prefixed) with the specified prefix.
+
+```graphql
+#::[Using the `@using` directive with a custom prefix]
+schema @using(spec: "https://spec.example.com/example/v1", prefix: "eg") {
+  query: Query
+}
+
+type User {
+  # Specifying `prefix: "eg"` transforms @example_someDirective from
+  # the example spec into @eg_someDirective
+  name: String @eg_someDirective
+}
+
+# Prefix transformation must also be applied to definitions pulled in from
+# specifications.
+directive @eg_someDirective on FIELD
+```
+
+This argument has no effect on specs which do not introduce prefixed schema elements.
```

Comment:
If I understand the above correctly, this should be "which do not introduce default-prefixed schema elements", right?

As in, if a spec introduces multiple prefixed schema elements, some of which use the spec's default prefix and some of which don't, then the above replacement process would only replace for the schema elements that used the default prefix, right? So then if a spec has a default prefix that's never used in any of its introduced prefixed schema elements, then it would also be unaffected presumably.

Reply:
Mmmm... we just refer to them as "prefixed schema elements" elsewhere in the doc. In the spec text they have the default prefix; in the document they have whatever is specified in their `@using(prefix:)` or the default prefix if there isn't one.

I can think of two cases where this is meaningful: 
  1. specs which do not introduce any schema elements at all, i.e. a spec which is effectively a flag telling someone something about the document
  2. specs which introduce schema elements which should carry a prefix, but don't. We explicitly say that specs shouldn't do this. If they do anyway, then yes, those would be unaffected.